model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/DesignElementHandle.java

```diff
@@ -1639,7 +1631,7 @@ public void copyPropertyTo(String propName, DesignElementHandle targetHandle) th
 	 * @see ErrorDetail
 	 */
 
-	public List semanticCheck() {
+	public List<?> semanticCheck() {
```

Comment:
This method should really be like this:
```
	public List<ErrorDetail> semanticCheck() {
		// Validate this element.

		List<SemanticException> exceptionList = getElement().validate(module);
		List<ErrorDetail> errorDetailList = ErrorDetail.convertExceptionList(exceptionList);

		return errorDetailList;
	}
```

But then the ErrorDetail method should be more accepting on any type of list of exceptions like this;
```
	public static List<ErrorDetail> convertExceptionList(List<? extends Exception> exceptionList) {
		List<ErrorDetail> errorDetailList = new ArrayList<>();

		Iterator<? extends Exception> iterError = exceptionList.iterator();
		while (iterError.hasNext()) {
			Exception e = iterError.next();

			ErrorDetail errorDetail = new ErrorDetail(e);
			errorDetailList.add(errorDetail);
		}

		return errorDetailList;
	}
```

Reply:
I haven't changed it because it would cause more changes according to your description of "convertExceptionList()" therefore I haven't changed all.

Reply:
I can try it but if there no success I will full reset the change back to version with warnings.

Reply:
Note that the return type of List<ErrorDetail> errorDetailList = ErrorDetail.convertExceptionList(exceptionList) is already correct and can be used, even if leave exceptionList  as a raw type.